deepinv/physics/generator/downsampling.py

```diff
@@ -88,17 +89,36 @@ def step(
         """
         self.rng_manual_seed(seed)
 
-        random_indices = torch.randint(
+        factor_indices = torch.randint(
             low=0,
             high=len(self.list_factors),
-            size=(2,),
+            size=(batch_size,),
             generator=self.rng,
             **self.factory_kwargs,
         )
-        factor = self.list_factors[int(random_indices[0].item())]
-        filter_str = self.list_filters[int(random_indices[1].item())]
-        filters = self.get_kernel(filter_str, factor)
+        filter_indices = torch.randint(
+            low=0,
+            high=len(self.list_filters),
+            size=(batch_size,),
+            generator=self.rng,
+            **self.factory_kwargs,
+        )
+        factors = torch.Tensor(
+            [self.list_factors[int(i)] for i in factor_indices.tolist()]
+        )
```

Comment:
`torch.Tensor` will always return a FloatTensor, I'm not sure if it is intended
```suggestion
        factors = torch.tensor(self.list_factors, device=self.device)[factor_indices]
```

Reply:
Or using directly `dinv.physics.functional.rand.random_choice`